app/src/main/AndroidManifest.xml

```diff
@@ -28,11 +28,16 @@
 
         <!-- Disable WorkManager's default initialization so we can use Dagger with it -->
         <provider
-            android:name="androidx.work.impl.WorkManagerInitializer"
-            android:authorities="${applicationId}.workmanager-init"
-            android:enabled="false"
+            android:name="androidx.startup.InitializationProvider"
+            android:authorities="${applicationId}.androidx-startup"
             android:exported="false"
-            tools:replace="android:authorities" />
+            tools:node="merge">
+            <!-- If you are using androidx.startup to initialize other components -->
+            <meta-data
+                android:name="androidx.work.WorkManagerInitializer"
+                android:value="androidx.startup"
+                tools:node="remove" />
+        </provider>
```

Comment:
Just curious, why do we need to add the `metada` now?

Reply:
It is in the documentation (https://developer.android.com/jetpack/androidx/releases/work), I actually tried without it and didn't work.